src/H5private.h

```diff
@@ -497,8 +497,16 @@
 #define H5_DIAG_DO_PRAGMA(x)  _Pragma(#x)
 #define H5_DIAG_PRAGMA(x)     H5_DIAG_DO_PRAGMA(GCC diagnostic x)
 
+/* Allow suppression of compiler diagnostics unless H5_SHOW_ALL_WARNINGS is
```

Comment:
When the 'show all warnings' build option is used, don't suppress compiler diagnostics.